pkg/providers/cloudstack/cloudstack.go

```diff
@@ -574,8 +577,10 @@ func (p *cloudstackProvider) needsNewKubeadmConfigTemplate(workerNodeGroupConfig
 // AnyImmutableFieldChanged Used by EKS-A controller and CLI upgrade workflow to compare generated CSDC/CSMC's from
 // CAPC resources in fetcher.go with those already on the cluster when deciding whether or not to generate and apply
 // new CloudStackMachineTemplates
-func AnyImmutableFieldChanged(oldCsdc, newCsdc *v1alpha1.CloudStackDatacenterConfig, oldCsmc, newCsmc *v1alpha1.CloudStackMachineConfig) bool {
+func AnyImmutableFieldChanged(oldCsdc, newCsdc *v1alpha1.CloudStackDatacenterConfig, oldCsmc, newCsmc *v1alpha1.CloudStackMachineConfig, log logr.Logger) bool {
```

Comment:
When looking at the code, passing the logger around just to get here and log any possible difference, it makes me think this might be information we want to return from this method. Instead of being just a boolean, return something like `EksaCloudstackDiff` and then let the caller decide what to do with that info (log it, pass it around, transform it, ignore it...).

No need to address as part of this PR since I understand we want to have this logs as soon as possible. But we should probably move into that direction.